src/core/p5.Renderer.js

```diff
@@ -60,6 +60,56 @@ p5.Renderer = function(elt, pInst, isMainCanvas) {
 
 p5.Renderer.prototype = Object.create(p5.Element.prototype);
 
+// the renderer should return a 'style' object that it wishes to
+// store on the push stack.
+p5.Renderer.prototype.push = function() {
+  return {
+    properties: {
+      _doStroke: this._doStroke,
+      _strokeSet: this._strokeSet,
+      _doFill: this._doFill,
+      _fillSet: this._fillSet,
+      _tint: this._tint,
+      _imageMode: this._imageMode,
+      _rectMode: this._rectMode,
+      _ellipseMode: this._ellipseMode,
+      _textFont: this._textFont,
+      _textLeading: this._textLeading,
+      _textSize: this._textSize,
+      _textStyle: this._textStyle
+    }
+  };
+};
+
+// this is implementation of Object.assign()
+// The assign() method is used to copy the values of all enumerable
+// own properties from one or more source objects to a target object.
+// It will return the target object.
+function assign(to, firstSource) {
```

Comment:
hmm, what's the reason to re-implement here instead of just using Object.assign directly?

Reply:
because Object.assign isn't widely supported. i initially used a polyfill from npm, but it had issues with global/instance modes

Reply:
Seems like only IE 11 doesn't support? http://kangax.github.io/compat-table/es6/#test-Object_static_methods_Object.assign

Reply:
on the desktop yeah, but many mobile devices don't support it either. i'm not sure this is really an issue. if i had just included the for loop in the pop() method, would this have even come up?

Reply:
while working on the GL code over last summer, I came across a number of stale + unused methods that had been left hanging in the code because of a refactor. I just want it to to be obvious where + why methods in the global scope are being used so that they don't get left around in future changes. using a built in method would also meet this goal; having a name of the method that makes its intended use/reach would also meet that goal.

Reply:
oh i absolutely agree. i just think it's a little too soon to add a dependency on Object.assign. i named the function the same name and implemented it in a compatible way so it'll be a more obvious replacement when, down the line, down-level browser compatibility for this method becomes less of an issue, or the browser compatibility requirements change.

i think this is a bigger issue, though. maybe it's worth having a document somewhere that specifically calls out what browsers/platforms the library is intended to support and then decisions like this can be made against that. i think a few extra bytes of zipped .js is a reasonable price to pay for improved comaptability.

Reply:
we do have a list of supported browsers, https://github.com/processing/p5.js/wiki/Supported-browsers @limzykenneth should we update this for 2018? do you think it's safe to use Object.assign?

Reply:
Yes that list is pretty outdated and can use some updates. As far as I can find, only Android web browser and Opera for Android don't support `Object.assign` which accounts for maybe ~2% of browser market (most of which are probably really outdated Androids), for me it's not a big enough share to worry about and unlike other less popular browsers (say Edge), there doesn't seem to be any sign that they will be updated to follow new standards. I'm also considering removing some of the existing polyfills like the `promise` polyfill from the library as we move to support newer browsers.

However, if it is decided that we will still keep the polyfill for now, I would suggest making some modification or just use the one on [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) which throws similar errors to native implementation and also attaches directly to the `Object`...object. That can make our future transition from polyfill to no polyfill smoother.

Reply:
can we delay that to another issue/PR to address in the future?

Reply:
My initial comments (now rendered outdated in the PR) asked for clarity on where this method is expected to be used and to declutter the global/module namespace, which makes using Object.assign a perfectly reasonable solution. Reducing the amount of custom code we have to maintain in favor of a built-in method seems like a win.

Reply:
i'd rather close this PR than kill compatibilty with IE & android devices.

the 25 bytes just aren't worth it.

Reply:
I might have misunderstood something, but this wouldn't kill compatibility with Android devices and it is also supported by Edge, so I think we're just talking about IE 11. The unsupported Android browser that Kenneth mentioned being <2% of the market is only on older devices and is longer the default browser (it is now Chrome). Are you sure you still want to withdraw on this point?

I'd also be okay with updating the comment to say that this is a polyfill & why it's used instead of Object.assign. I just want the maintenance to be easy to track.